cpp/include/cudf/fixed_point/fixed_point.hpp

```diff
@@ -315,8 +315,7 @@ class fixed_point {
    * @brief Default constructor that constructs `fixed_point` number with a
    * value and scale of zero
    */
-  CUDA_HOST_DEVICE_CALLABLE
-  fixed_point() : _value{0}, _scale{scale_type{0}} {}
+  fixed_point() = default;
```

Comment:
A type used in shared memory cannot perform _any_ initialization, i.e., it must be trivially constructible. The fact that this happens to compile is a compiler bug.

Reply:
I guess I'm reading the dev guide differently:

> A __device__, __constant__, __managed__ or __shared__ variable definition cannot have a class type with a non-empty constructor or a non-empty destructor. A constructor for a class type is considered empty at a point in the translation unit, if it is either a trivial constructor or it satisfies all of the following conditions:
> - The constructor function has been defined.
> - The constructor function has no parameters, the initializer list is empty and the function body is an empty compound-statement.
> - Its class has no virtual functions and no virtual base classes.
> - The default constructors of all base classes of its class can be considered empty.
> - For all the nonstatic data members of its class that are of class type (or array thereof), the default constructors can be considered empty.

The last bullet implies that non-static members must be considered empty which takes us back to checking all the bullets for that member which looks true to me.

Perhaps there is a subtle implication that the member variable cannot have any initializer `{}` defined regardless?

Reply:
I checked with the compiler folks and the constructor definitely isn't legal, we just aren't seeing a diagnostic.

Furthermore, I believe the documentation is incorrect/lacking. I filed an nvbug for both of these things: get the compiler to emit a diagnostic, update the CUDA docs.

Reply:
It's strange that its just a warning. I guess it knows how to create the memory for it so it does not stop generating code.